pkg/server/backend_manager.go

```diff
@@ -121,17 +120,39 @@ func (b *backend) Context() context.Context {
 	return b.conn.Context()
 }
 
-func newBackend(conn agent.AgentService_ConnectServer) *backend {
+func (b *backend) GetAgentIdentifiers() (header.Identifiers, error) {
```

Comment:
optional: consider doing this work when `NewBackend(...)` is called, and caching the agent identifiers in the `backend` struct instead.